### PR TITLE
[IMP] crm: improve pipeline analysis report search view

### PR DESCRIPTION
**Pull request by ija-odoo**
PURPOSE

Improve the pipeline analysis search for a better usability.

SPECIFICATIONS

The goal of this commit is to improve the pipeline analysis search
view for a better usability, so the main changes going to be done
in this commit is reordering,renaming,removing and adding some
filters in the pipeline analysis report search view.

As we need to display same filters and groupby in both community and
enterprise pipeline report search view, we have removed the search
view `crm_opportunity_view_search` from enterprise so that we can use
the community view there and both becomes the same view.

So in this commit we have also added some filters and groupby to
community version(`crm.crm_opportunity_report_view_search`) which was
existed in enterprise search view (`crm_enterprise.crm_opportunity_view_search`)
so that we dont miss any filters used in enterprise.
        
LINKS
PR #71796
Task 2343223

### CODE REVIEW

File: addons/crm/report/crm_opportunity_report_views.xml [hunk 1]
```diff
@@ -61,11 +61,11 @@
             <field name="priority">32</field>
             <field name="arch" type="xml">
                 <search string="Opportunities Analysis">
-                    <filter string="My Opportunities" name="my"
+                    <filter string="My Pipeline" name="my"
                             domain="[('user_id', '=', uid)]"/>
                     <separator/>
-                    <filter name="lead" string="Lead" domain="[('type','=', 'lead')]" help="Show only lead"/>
-                    <filter name="opportunity" string="Opportunity" domain="[('type','=','opportunity')]" help="Show only opportunity"/>
+                    <filter name="opportunity" string="Opportunities" domain="[('type','=','opportunity')]" help="Show only opportunity" groups="crm.group_use_lead"/>
```
Comment (tde-banana-odoo) [inline addons/crm/report/crm_opportunity_report_views.xml:67]: Maybe "show only opportunities", as we probably have more than 1 ?

File: addons/crm/report/crm_opportunity_report_views.xml [hunk 2]
```diff
@@ -61,11 +61,11 @@
...
+                    <filter name="lead" string="Leads" domain="[('type','=', 'lead')]" help="Show only lead" groups="crm.group_use_lead"/>
```
Comment (tde-banana-odoo) [inline addons/crm/report/crm_opportunity_report_views.xml:68]: Same here, "show only leads" ?

**Review by tde-banana-odoo — COMMENTED** [review]
Validation: wording suggestion

File: addons/crm/report/crm_opportunity_report_views.xml (hunk 1 above)
Comment (tde-banana-odoo) [reply addons/crm/report/crm_opportunity_report_views.xml:67]: Note that we could also rename the filter to match guidelines (filter_opportunity)

File: addons/crm/report/crm_opportunity_report_views.xml (hunk 2 above)
Comment (tde-banana-odoo) [reply addons/crm/report/crm_opportunity_report_views.xml:68]: Note that we could also rename the filter to match guidelines (filter_lead)

**Review by awa-odoo — COMMENTED** [review]
Hello @ija-odoo 

Few remarks about the test but not much to say otherwise.

But also:
- Shouldn't the fix be targeting 15.0?
  It's mentioning a commit that introduced the issue in 15.0 so I'm a bit confused
- Same as enterprise, can you add the enterprise and upgrade PRs links in your main commit?

Thanks!

**Pull request closed without merging**